openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -415,6 +794,48 @@ impl PagePoolAllocator {
             size_pages,
         })
     }
+
+    /// Restore allocations after a restore operation on the pool. This will
+    /// return any allocations that were previously allocated for this
+    /// allocator.
+    pub fn restore_allocations(&self) -> Vec<PagePoolHandle> {
```

Comment:
I had assumed the devices would restore things one at a time and not just get a batch of restored allocations. is this not what we want?

Reply:
I must have misunderstood what you proposed then. Do we want each device to call alloc again and get a restored allocation, or what did you imagine?

Reply:
They should call `.restore()`.

Reply:
Okay easy enough.